extensions/interactions/MathEquationInput/directives/oppia-interactive-math-equation-input.component.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { WindowRef } from 'services/contextual/window-ref.service.ts';
+
+require(
+  'interactions/MathEquationInput/directives/' +
+  'math-equation-input-rules.service.ts');
+require(
+  'pages/exploration-player-page/services/current-interaction.service.ts');
+require(
+  'interactions/MathEquationInput/directives/' +
+  'oppia-interactive-math-equation-input.component.ts');
+
+describe('MathEquationInputInteractive', function() {
+  let ctrl = null, $window = null;
+  let mockCurrentInteractionService = {
+    onSubmit: function(answer, rulesService) {},
+    registerCurrentInteraction: function(submitAnswerFn, validateEquationFn) {
+      submitAnswerFn();
+    }
+  };
+  let mockMathEquationInputRulesService = {};
+  let mockGuppyObject = {
+    guppyInstance: {
+      asciimath: function() {
+        return 'Dummy value';
+      }
+    }
+  };
+  let guppyConfigurationService = null;
+  let mathInteractionsService = null;
+  let guppyInitializationService = null;
+
+  class MockGuppy {
+    constructor(id: string, config: Object) {}
+
+    asciimath() {
+      return 'Dummy value';
+    }
+    static event(name: string, handler: Function): void {
+      handler();
+    }
+    static configure(name: string, val: Object): void {}
+    static 'remove_global_symbol'(symbol: string): void {}
+  }
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    guppyConfigurationService = new GuppyConfigurationService(
+      new DeviceInfoService(new WindowRef()));
+    mathInteractionsService = new MathInteractionsService();
+    guppyInitializationService = new GuppyInitializationService();
+
+    $provide.value('CurrentInteractionService',
+      mockCurrentInteractionService);
+    $provide.value('MathEquationInputRulesService',
+      mockMathEquationInputRulesService);
+    $provide.value('GuppyConfigurationService', guppyConfigurationService);
+    $provide.value('MathInteractionsService', mathInteractionsService);
+    $provide.value('GuppyInitializationService', guppyInitializationService);
+  }));
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $window = $injector.get('$window');
+    ctrl = $componentController('oppiaInteractiveMathEquationInput');
+    $window.Guppy = MockGuppy;
+  }));
+
+  it('should add the change handler to guppy', function() {
+    spyOn(guppyInitializationService, 'findActiveGuppyObject').and.returnValue(
+      mockGuppyObject);
+    ctrl.$onInit();
+    expect(guppyInitializationService.findActiveGuppyObject).toHaveBeenCalled();
+  });
+
+  it('should not submit the answer if invalid', function() {
+    ctrl.hasBeenTouched = true;
+    // Invalid answer.
+    ctrl.value = '(x + y)) = 3';
+
+    spyOn(mockCurrentInteractionService, 'onSubmit');
+    ctrl.submitAnswer();
+    expect(mockCurrentInteractionService.onSubmit).not.toHaveBeenCalled();
```

Comment:
Is there some error message displayed? If so can we assert that?

Reply:
Yes, I've added it here and in the algebraic component.